middleman-core/lib/middleman-core/preview_server.rb

```diff
@@ -205,9 +207,19 @@ def setup_webrick(is_logging)
         end
 
         begin
+          tries ||= 4
+          tried_ports ||= []
+
           ::WEBrick::HTTPServer.new(http_opts)
         rescue Errno::EADDRINUSE
-          logger.error "== Port #{port} is unavailable. Either close the instance of Middleman already running on #{port} or start this Middleman on a new port with: --port=#{port.to_i + 1}"
+          unless (tries -= 1).zero?
```

Comment:
I'd personally prefer if this were two lines:

```ruby
tries -= 1
if tries > 0
```

`tries` is also a sort of odd name since it counts down - maybe `tries_remaining`?